api/debug.go

```diff
@@ -22,77 +39,371 @@ type txTraceResult struct {
 }
 
 type DebugAPI struct {
-	logger    zerolog.Logger
-	tracer    storage.TraceIndexer
-	blocks    storage.BlockIndexer
-	collector metrics.Collector
+	registerStore *pebble.RegisterStorage
+	logger        zerolog.Logger
+	tracer        storage.TraceIndexer
+	blocks        storage.BlockIndexer
+	transactions  storage.TransactionIndexer
+	receipts      storage.ReceiptIndexer
+	client        *requester.CrossSporkClient
+	config        *config.Config
+	collector     metrics.Collector
 }
 
-func NewDebugAPI(tracer storage.TraceIndexer, blocks storage.BlockIndexer, logger zerolog.Logger, collector metrics.Collector) *DebugAPI {
+func NewDebugAPI(
+	registerStore *pebble.RegisterStorage,
+	tracer storage.TraceIndexer,
+	blocks storage.BlockIndexer,
+	transactions storage.TransactionIndexer,
+	receipts storage.ReceiptIndexer,
+	client *requester.CrossSporkClient,
+	config *config.Config,
+	logger zerolog.Logger,
+	collector metrics.Collector,
+) *DebugAPI {
 	return &DebugAPI{
-		logger:    logger,
-		tracer:    tracer,
-		blocks:    blocks,
-		collector: collector,
+		registerStore: registerStore,
+		logger:        logger,
+		tracer:        tracer,
+		blocks:        blocks,
+		transactions:  transactions,
+		receipts:      receipts,
+		client:        client,
+		config:        config,
+		collector:     collector,
 	}
 }
 
-// TraceTransaction will return a debug execution trace of a transaction if it exists,
-// currently we only support CALL traces, so the config is ignored.
+// TraceTransaction will return a debug execution trace of a transaction, if it exists.
 func (d *DebugAPI) TraceTransaction(
 	_ context.Context,
 	hash gethCommon.Hash,
-	_ *tracers.TraceConfig,
+	config *tracers.TraceConfig,
 ) (json.RawMessage, error) {
-	res, err := d.tracer.GetTransaction(hash)
+	// If the given trace config is equal to the default call tracer used
+	// in block replay during ingestion, then we fetch the trace result
+	// from the Traces DB.
+	if isDefaultCallTracer(config) {
+		trace, err := d.tracer.GetTransaction(hash)
```

Comment:
Lets log the error.

In the future (follow up issue) we should differentiate  between err "trace not found" and any other error. I trace was not found, just compute it now, if its a different error, then error out.

Reply:
Good idea :+1: 
Added in https://github.com/onflow/flow-evm-gateway/pull/635/commits/dc4d6cc21fd99c3ed2c61aa57215c1c155e73017